core/src/services/routing.js

```diff
@@ -98,11 +98,29 @@ export const isNotSameDomain = (config, component) => {
       componentData.previousNodeValues.viewUrl
     );
     const nextUrl = getUrlWithoutHash(componentData.viewUrl);
-    return previousUrl != nextUrl;
```

Comment:
I will refactor the function so it doesn't have negation, but I don't like the `isSameViewGroup` nam, because in hash routing we don't have viewGroups. Maybe something more like `isSameApplication`, `isSameIFrame`?